src/data_cell/sparse_graph_datacell.cpp

```diff
@@ -0,0 +1,95 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "sparse_graph_datacell.h"
+
+namespace vsag {
+
+SparseGraphDataCell::SparseGraphDataCell(const JsonType& graph_json_params,
+                                         const IndexCommonParam& common_param)
+    : allocator_(common_param.allocator_), neighbors_(allocator_) {
```

Comment:
minor: Here depends the fields initialize sequence
can change to: neighbors_(common_param.allocator_)

Reply:
done